google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerImpl.java

```diff
@@ -279,6 +279,7 @@ public DatabaseClient getDatabaseClient(DatabaseId db) {
                 labelValues,
                 attributesBuilder.build());
         pool.maybeWaitOnMinSessions();
+        pool.waitOnMultiplexedSession();
```

Comment:
Note that we are in an `synchronized (this)` block here, and that (in the current implementation) blocking is the default behavior. That means that we are blocking here while also holding a lock. That also means that multiple concurrent calls to `Spanner.getDatabaseClient(..)` will be serialized and wait for each other's multiplexed session to be created.
(Preferably, we should also move the `pool.maybeWaitOnMinSessions()` call to outside the block.)